File: src/test/java/fr/deroffal/aoc/Day02Test.java
package fr.deroffal.aoc;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Day02Test {

    @Test
    void input5195_retourne_8() {
        assertEquals(8, Day02.calculerDifference("5\t1\t9\t5"));
    }

    @Test
    void input753_retourne_4() {
        assertEquals(4, Day02.calculerDifference("7\t5\t3"));
    }

    @Test
    void input2468_retourne_6() {
        assertEquals(6, Day02.calculerDifference("2\t4\t6\t8"));
    }

    @Test
    void input5928_retourne_4() {
        assertEquals(4, Day02.calculerDifferenceDivision("5\t9\t2\t8"));
    }

    @Test
    void input9473_retourne_3() {
        assertEquals(3, Day02.calculerDifferenceDivision("9\t4\t7\t3"));
    }

    @Test
    void input3865_retourne_2() {
        assertEquals(2, Day02.calculerDifferenceDivision("3\t8\t6\t5"));
    }

}

File: kotlin/src/test/kotlin/fr/deroffal/aoc/kotlin/Day02Test.kt
package fr.deroffal.aoc.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class Day02Test {

    private val day02 = Day02()

    @Test
    fun `part1 matches examples`() {
        val input: List<String> = listOf("5 1 9 5", "7 5 3", "2 4 6 8")
        assertEquals(18, day02.solvePart1(input))
    }

    @Test
    fun `part2 matches examples`() {
        val input: List<String> = listOf("5 9 2 8", "9 4 7 3", "3 8 6 5")
        assertEquals(9, day02.solvePart2(input))
    }

}
